presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/AbstractTestIcebergSmoke.java

```diff
@@ -1389,4 +1395,55 @@ private void dropTable(String table)
         assertUpdate(session, "DROP TABLE " + table);
         assertFalse(getQueryRunner().tableExists(session, table));
     }
+
+    @Test
+    public void testLocalDynamicFilterWithEmptyBuildSide()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                withBroadcastJoin(),
+                "SELECT * FROM lineitem JOIN supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'abc'");
+        assertEquals(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "tpch.lineitem");
+        assertEquals(probeStats.getInputPositions(), 0);
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), probeStats.getTotalDrivers());
+    }
+
+    @Test
+    public void testDynamicFilterWithSelectiveBuildSide()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                withBroadcastJoin(),
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.totalprice = 172799.49");
+        assertGreaterThan(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "tpch.lineitem");
+        // Probe-side is partially scanned
+        assertLessThan(probeStats.getInputPositions(), 60175L);
+        assertEquals(probeStats.getDynamicFilterSplitsProcessed(), probeStats.getTotalDrivers());
+    }
+
+    @Test
+    public void testDynamicFilterWithNonSelectiveBuildSide()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                withBroadcastJoin(),
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey");
```

Comment:
Please use `lineitem.suppkey = supplier.suppkey` instead.
In this case the dynamic filter is set to `all` because the build side is too big, whereas we want to test the case where there is a dynamic filter but it doesn't filter anything.